### DISCUSSION TITLE
node20 as default version

### DISCUSSION THREAD

**Discussion by yanivpaz** [actions/runner#3430] [General] [answered]
Hi all 
following https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/
1.
any idea why node20 is not the default ?
[https://github.com/actions/runner/blob/main/src/Misc/layoutbin/runsvc.sh - L13](https://github.com/actions/runner/blob/main/src/Misc/layoutbin/runsvc.sh#L13)

2. Wouldn't it will be better to pack the action runner with relevant node version only ?
```
actions-runner/externals
90M     node16_alpine
98M     node20_alpine
103M    node16
165M    node20
```

**Comment by nebuk89** [comment, marked as answer]
Hey! We keep node16 on there so folks can feature flag back if their OS doesn't support node20 as part of our migration path. We will be removing node16 no later than November 2024
